src/ORM/FieldType/DBYear.php

```diff
@@ -25,22 +36,62 @@ public function scaffoldFormField(?string $title = null, array $params = []): ?F
         return $selectBox;
     }
 
+    public function setValue(mixed $value, null|array|ModelData $record = null, bool $markChanged = true): static
+    {
+        parent::setValue($value, $record, $markChanged);
+        // 0 is used to represent a null value, which will be stored as 0000 in MySQL
+        if ($this->value === '0000') {
+            $this->value = 0;
+        }
+        // shorthand for 2000 in MySQL
+        if ($this->value === '00') {
+            $this->value = 2000;
+        }
+        // convert string int to int
+        // string int and int are both valid in MySQL, though only use int internally
+        if (is_string($this->value) && preg_match('#^\d+$#', (string) $this->value)) {
+            $this->value = (int) $this->value;
+        }
+        if (!is_int($this->value)) {
+            return $this;
+        }
+        // shorthand for 2001-2069 in MySQL
+        if ($this->value >= 1 && $this->value <= 69) {
+            $this->value = 2000 + $this->value;
+        }
+        // shorthand for 1970-1999 in MySQL
+        if ($this->value >= 70 && $this->value <= 99) {
+            $this->value = 1900 + $this->value;
+        }
+        return $this;
+    }
+
+    public function getMinYear(): int
+    {
+        return DBYear::MIN_YEAR;
+    }
+
+    public function getMaxYear(): int
+    {
+        return DBYear::MAX_YEAR;
+    }
+
     /**
      * Returns a list of default options that can
      * be used to populate a select box, or compare against
      * input values. Starts by default at the current year,
-     * and counts back to 1900.
+     * and counts back to 1901.
```

Comment:
```suggestion
     * and counts back to the minimum value supported by the connected database server.
```

Reply:
I've left it as is as I've noted this class is MySQL specific